### ISSUE TITLE
Forward-reverse grapheme mismatch on "\u{1F938}\u{1F3FE}\u{1F3FE}"

### ISSUE THREAD

**Issue by Manishearth** [unicode-rs/unicode-segmentation#19]
Found by [cargo-fuzz](https://github.com/rust-fuzz/cargo-fuzz).

This crashes:

```

extern crate unicode_segmentation;

use unicode_segmentation::UnicodeSegmentation;
fn main() {
    let s = "\u{1F938}\u{1F3FE}\u{1F3FE}";
    let forward = UnicodeSegmentation::graphemes(s, true).collect::<Vec<_>>();
    let forward_reversed = forward.into_iter().rev().collect::<Vec<_>>();
    let reverse = UnicodeSegmentation::graphemes(s, true).rev().collect::<Vec<_>>();
    assert_eq!(forward_reversed, reverse);
}

```

It panics with (manually escaped):

```
left `["\u{1F938}\u{1F3FE}\u{1F3FE}"]`, right: `["\u{1F3FE}", "\u{1F938}\u{1F3FE}"]`
```

The original emoji is ([uniview](https://r12a.github.io/uniview/?charlist=%F0%9F%A4%B8%F0%9F%8F%BE%F0%9F%8F%BE)) U+1F938 PERSON DOING CARTWHEEL followed by two Fitzpatrick skin color modifiers. I suspect this error will happen whenever you have two skin color modifiers after a modifiable emoji.

cc @mbrubeck  @kwantam

**Comment by Manishearth** [comment]
There's another mismatch with `jjjjjjjjj0'jjjj4(\u{0489}:4\u{FFFD}\u{FFFD}_\u{200D}\u{2764}_'\u{0589}` (base64: `ampqampqampqMCdqampqNCjSiTo0AABf4oCN4p2kXyfWiQ==`)

**Comment by Manishearth** [comment]
We have another mismatch on `\u{200D}\u{200D}\u{2764}\u{2764}`; the forward iterator considers it a single grapheme but the reverse iterator splits off the last heart into its own grapheme. This is likely a separate bug, but I'm listing it here.

**Comment by HadrienG2** [comment]
Should be fixed by #21?
